qemu/tests/hotplug_mem.py

```diff
@@ -31,40 +31,35 @@ def run_background_test(self):
         args = (self.test, self.params, self.env, self.params["sub_type"])
         bg_test = BackgroundTest(run_virt_sub_test, args)
         bg_test.start()
-        wait_for(bg_test.is_alive, timeout=60)
+        wait_for(bg_test.is_alive, first=10, step=3, timeout=100)
         return bg_test
 
-    def restore_memory(self, origin_vm, post_vm):
-        mem_devs_post = post_vm.params.objects("mem_devs")
-        mem_devs_origin = origin_vm.params.objects("mem_devs")
-        if len(mem_devs_post) != len(mem_devs_origin):
-            if len(mem_devs_origin) > len(mem_devs_post):
-                mem_devs = set(mem_devs_origin) - set(mem_devs_post)
-                vm, operation = post_vm, "hotplug"
-            else:
-                mem_devs = set(mem_devs_post) - set(mem_devs_origin)
-                vm, operation = origin_vm, "unplug"
-            func = getattr(self, "%s_memory" % operation)
-            for mem_dev in mem_devs:
-                func(vm, mem_dev)
-                self.check_vm_memory(vm)
+    def restore_memory(self, pre_vm, post_vm):
+        mem_devs_post = set(post_vm.params.objects("mem_devs"))
+        mem_devs_origin = set(pre_vm.params.objects("mem_devs"))
+        if mem_devs_post == mem_devs_origin:
+            return
+        if len(mem_devs_origin) > len(mem_devs_post):
+            mem_devs = mem_devs_origin - mem_devs_post
+            vm, operation = post_vm, "hotplug"
+        elif len(mem_devs_origin) < len(mem_devs_post):
+            mem_devs = mem_devs_post - mem_devs_origin
+            vm, operation = pre_vm, "unplug"
+        func = getattr(self, "%s_memory" % operation)
+        map(lambda x: func(vm, x), mem_devs)
 
     def get_mem_by_name(self, vm, name):
         """
         Return memory object and pc-dimm devices by given name
         """
-        devices = []
-        for dtype in ["mem", "dimm"]:
-            dev_qid = "%s-%s" % (dtype, name)
-            devs = vm.devices.get_by_qid(dev_qid)
-            if devs:
-                devices.append(devs[0])
-        return devices
+        dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
```

Comment:
00:52:17 ERROR| Traceback (most recent call last):
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 169, in runTest
00:52:17 ERROR|     self._runTest()
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/avocado_vt/test.py", line 311, in _runTest
00:52:17 ERROR|     run_func(self, params, env)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/avocado-vt/virttest/error_context.py", line 135, in new_fn
00:52:17 ERROR|     return fn(*args, **kwargs)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 108, in run
00:52:17 ERROR|     hotplug_test.start_test()
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 68, in start_test
00:52:17 ERROR|     devs = self.get_mem_by_name(vm, target_mem)
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 55, in get_mem_by_name
00:52:17 ERROR|     dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
00:52:17 ERROR|   File "/home/staf-kvm-devel/workspace/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/hotplug_mem.py", line 55, in <lambda>
00:52:17 ERROR|     dev_ids = map(lambda x: "-".join(x, name), ["mem", "dimm"])
00:52:17 ERROR| TypeError: join() takes exactly one argument (2 given)